### PR TITLE
Local Market Analysis

### PR DESCRIPTION
**Pull request by Evanito**
## Description

The bot will record and analyse lending market data to dynamically change the `minlendrate` to an effective rate for each coin's market, based entirely off of your aggressiveness setting.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] **I have read CONTRIBUTING.md**
- [x] **I fully understand [Github Flow.](https://guides.github.com/introduction/flow/)**
- [x] **My code adheres to the [code style of this project.](https://poloniexlendingbot.readthedocs.io/en/latest/contributing.html)**
- [x] **I have updated the documentation in /docs if I have changed the config, arguments, logic in how the bot works, or anything that understandably needs a documentation change.**
- [x] **I have updated the config file accordingly if my change requires a new configuration setting or changes an existing one.**
- [x] **I have tested the bot with no issues for 24 continuous hours. If issues were experienced, they have been patched and tested again.**

Currently: records market data, then updates the minimum rate a currency will lend at based on its historical performance. This is a Cherrypicked commit because the old branch was based on a merged fork.

### CODE REVIEW

File: default.cfg.example [hunk 1]
```diff
@@ -68,11 +68,22 @@ keepstuckorders = True
...
+analyseMaxAge = 30
+# Choose how often to record a market's rate. (In seconds.) (10-3600)
+analyseUpdateInterval = 60
+# Choose how aggressive to lend at based on a market's performance. (As a percentile of the market's rate.) (1-99)
+# Recommendations: Conservative = 50, Moderate = 75, Aggressive = 90, Very Aggressive = 99
+lendingStyle = 75
+
+#Syntax: ["COIN:mindailyrate:maxactiveamount:maxtolent:maxpercenttolent:maxtolentrate",...]
+#If maxactive amount is 0: stop lending this coin. in the future you'll be able to limit amount to be lent.
+#If maxtolent is 0: check for maxpercenttolent.
```
Comment (rnevet) [inline default.cfg.example]: restored *lent* mispelling

File: default.cfg.example [hunk 2]
```diff
@@ -68,11 +68,22 @@ keepstuckorders = True
...
 #maxtolendrate = 0
 
-#Syntax: ["COIN:mindailyrate:maxactiveamount:maxtolend:maxpercenttolend:maxtolendrate",...]
-#If maxactive amount is 0: stop lending this coin. in the future you'll be able to limit amount to be lend.
-#If maxtolend is 0: check for maxpercenttolend.
-#If maxpercenttolend is 0: 100% is going to be lend.
-#If maxtolendrate is set to more than 0: the maxtolend or maxpercenttolend will be used when then rate is less or equal to the maxtolendrate. if set to 0 the bot will use the maxtolend or maxpercenttolend all the time.
+# Local Market Analysis: Choose which coins to record the market of in order to analyse them and compute a recommended lending rate.
+# All coins: STR,BTC,BTS,CLAM,DOGE,DASH,LTC,MAID,XMR,XRP,ETH,FCT
+#analyseCurrencies = STR,BTC,BTS,CLAM,DOGE,DASH,LTC,MAID,XMR,XRP,ETH,FCT
```
Comment (rnevet) [inline default.cfg.example]: Shouldn't this be automatically selected from coins in the lending account?

**Comment by Evanito** [comment]
@rnevet Been busy, will take those into account but this is not in a state to merge. Will remove DNM label once it is a candidate.

**Comment by rnevet** [comment]
@Evanito I know, comments are there for when you have time.

**Comment by Evanito** [comment]
I need to find a method to get a list of all currencies that are in the lending market. I already have one that returns a list of lending currencies the user has active.

File: default.cfg.example (hunk 1 above)
Comment (Evanito) [inline default.cfg.example]: Done

File: default.cfg.example (hunk 2 above)
Comment (Evanito) [inline default.cfg.example]: Done, made into setting.

**Comment by Evanito** [comment]
@rnevet What do you think of:
1. Allowing maxtolendrate to be use the new get_rate_suggestion() to have dynamic rates for each currency.
2. Showing some data about each currency in the web server (Would need to delegate that to you.)
3. Leaving the code "open" for alternative algorithms for getting a recommended rate from the market data.

**Comment by rnevet** [comment]
@yura-pakhuchiy  What do you think about this change? any ideas for the UI?

File: modules/Data.py [hunk 3]
```diff
@@ -9,6 +10,7 @@ def init(api1, log1):
     global api, log
     api = api1
     log = log1
+    get_lending_currencies()
```
Comment (rnevet) [inline modules/Data.py]: Data.init is called outside the exception handling loop and if this call fails - the exception won't be handled.

File: docs/configuration.rst [hunk 4]
```diff
@@ -176,21 +179,56 @@ This feature group allows you to only lend a certain percentage of your total ho
...
+---------------
+
+This feature allows you to record a currency's market and have the bot see trends. With this data, we can compute a recommended minimum lending rate per currency to avoid lending at times when the rate dips.
+
+``analyseCurrencies`` is the list of currencies to analyse.
+
+- Format: ``CURRENCY_TICKER,STR,BTC,BTS,CLAM,DOGE,DASH,LTC,MAID,XMR,XRP,ETH,FCT,ALL,ACTIVE``
+- Commenting it out will disable the entire feature.
+- Entering ``ACTIVE`` within the list will transfer any currencies that are found in your lending account, as well as any other currencies alongside it. Example: ``ACTIVE, BTC, CLAM`` will do BTC, CLAM, and any coins you are already lending.
+- Entering ``ALL`` will simply transfer all coins available to lending.
```
Comment (Evanito) [inline docs/configuration.rst]: Copy-paste error

File: docs/installation.rst [hunk 5]
```diff
@@ -34,49 +34,51 @@ To configure the bot with your settings:
...
 
 .. note:: If you are running out of CPU time every day: It is recommended to use a high sleeptimeinactive time for this website, as they meter your CPU usage.
 
-1.2.3 Creating the Web App (Optional)
--------------------------------------
+Creating the Web App (Optional)
+-------------------------------
 1. If you would like to use the Webserver to view your bot's status, navigate to the "Web" tab.
-2. Add a new web app.
+2. Add a new web app.i
```
Comment (Evanito) [inline docs/installation.rst]: Fix typo

File: modules/Data.py (hunk 3 above)
Comment (Evanito) [inline modules/Data.py]: Had this for debug, accidentally left it in.

**Pull request merged**